### ISSUE TITLE
Ensure Handlers leave the token stream in the correct spot and don't advance too far

### ISSUE THREAD

**Issue by SamWhited** [mellium/issuetest#33]
If a Handler returns an error mid way through parsing an element, the [`Serve`](https://godoc.org/mellium.im/xmpp#Serve) function (or other handlers that have subhandlers registered on them) needs to ensure that we skip the rest of the element.

Right now I don't see an easy way to do this, so it might require changes to how [`Handler`](https://godoc.org/mellium.im/xmpp#Handler) works. Eg. instead of passing the session, we might pass some sort of "token stream" that tracks where it is.

Furthermore, we need to ensure Handler's can't read beyond the end of the element which they were called to handle; some sort of similar token stream limiter API could also ensure this.

In short, we need a tokenizer in the [`mellium.im/xmlstream`](https://godoc.org/mellium.im/xmlstream) package that only allows tokenization of a single element (no more, no less) and an API proposal must be made for how this can be integrated and used with the current Serve/Handler APIs.
